File: csharp/ITEmployee.cs
using System;
using System.Text.Json.Serialization;

[Serializable]
public class ITEmployee : FullTimeEmployee
{
    [JsonInclude]
    private string? Specialization { get; set; }

    public ITEmployee() : base() { }

    public ITEmployee(string id, string firstName, string lastName, DateTime dob,
        string position, string department, double baseSalary, string benefits, string specialization)
        : base(id, firstName, lastName, dob, position, department, baseSalary, benefits)
    {
        if (department != "IT")
            throw new InvalidDataException("ITEmployee must belong to the IT department.");
        Specialization = specialization;
    }

    public override void ReportToManager()
    {
        if (Manager != null)
            Console.WriteLine($"{GetFullName()} (IT Employee, {Specialization}) reports to manager {Manager.GetFullName()}");
        else
            Console.WriteLine($"{GetFullName()} (IT Employee, {Specialization}) has no assigned manager.");
    }

    public void ShowSpecialization() => Console.WriteLine($"{GetFullName()}'s specialization: {Specialization}");
}

File: java/ITEmployee.java
import java.util.Date;

public class ITEmployee extends FullTimeEmployee {
    private String specialization;

    public ITEmployee(String id, String firstName, String lastName, Date dob,
            String position, String department, double baseSalary, String benefits,
            String specialization) throws InvalidDataException {
        super(id, firstName, lastName, dob, position, department, baseSalary, benefits);

        // Subclass-specific validation
        if (!"IT".equals(department)) {
            throw new InvalidDataException("ITEmployee must belong to the IT department.");
        }
        
        this.specialization = specialization;
    }

    @Override
    public void reportToManager() {
        if (manager != null) {
            System.out.println(getFullName() + " (IT Employee, " + specialization + ") reports to manager "
                    + manager.getFullName());
        } else {
            System.out.println(getFullName() + " (IT Employee, " + specialization + ") has no assigned manager.");
        }
    }

    public void showSpecialization() {
        System.out.println(getFullName() + "'s specialization: " + specialization);
    }
}